ejb3/src/main/java/org/jboss/as/ejb3/security/JaccInterceptor.java

```diff
@@ -0,0 +1,160 @@
+package org.jboss.as.ejb3.security;
+
+import static java.security.AccessController.doPrivileged;
+
+import java.lang.reflect.Method;
+import java.security.AccessController;
+import java.security.Policy;
+import java.security.PrivilegedAction;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.security.ProtectionDomain;
+
+import javax.security.jacc.EJBMethodPermission;
+import javax.security.jacc.PolicyContext;
+
+import org.jboss.as.ee.component.Component;
+import org.jboss.as.ee.component.ComponentView;
+import org.jboss.as.ejb3.component.EJBComponent;
+import org.jboss.as.ejb3.component.MethodIntf;
+import org.jboss.as.ejb3.logging.EjbLogger;
+import org.jboss.invocation.Interceptor;
+import org.jboss.invocation.InterceptorContext;
+import org.jboss.metadata.ejb.spec.MethodInterfaceType;
+import org.wildfly.security.manager.WildFlySecurityManager;
+
+/**
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class JaccInterceptor implements Interceptor {
+
+    private final String contextID;
+    private final String viewClassName;
+    private final Method viewMethod;
+
+    public JaccInterceptor(String contextID, String viewClassName, Method viewMethod) {
+        this.contextID = contextID;
```

Comment:
One thing to note is that the EJB subsystem currently contains another interceptor that also sets the policy context id (org.jboss.as.ejb3.security.PolicyContextIdInterceptor). May be able to remove that one now.